happi/cli.py

```diff
@@ -165,10 +165,18 @@ def happi_cli(args):
     elif args.cmd == 'edit':
         logger.debug('Starting edit block')
         device = client.find_device(name=args.name)
+        is_invalid_field = False
         for edit in args.edits:
             field, value = edit.split('=', 1)
-            logger.info(f'Setting {args.name}.{field} = {value}')
-            setattr(device, field, value)
+            try:
+                getattr(device, field)
+                logger.info(f'Setting {args.name}.{field} = {value}')
```

Comment:
`f`-strings are great, but they shouldn't be used here.

I know you copied the code that was there, but we should all make a point of fixing this when we see it.
`logger.debug()` et al take a format argument and a list of parameters to format, akin to perhaps `printf` in C.
It performs better (the format string isn't interpolated unless the log message is above the log filtering settings level) and won't crash your program if you messed up the string formatting somehow.

```suggestion
                logger.info('Setting %s.%s = %s', args.name, field, value)
```

Reply:
Ohh that's good to know, I did not know that. Thank you @klauer